### PR TITLE
Sync Insights ✨

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Add ability to view stats on un synced local changes on device

This feature can be accessed by clicking on the _Insights_ menu time under **Settings**
Insights_Screenshot

It should then render this view
Unsynced_Resources_Screen

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ndegwamartin** [comment]
It should be noted that there is a similar and more enhanced feature on the FHIR Core roadmap but we need this MVP to support more efficient sync verification and troubleshooting for QA and Devs and Users on the ground

**Comment by dubdabasoduba** [comment]
@ndegwamartin are this available for prod releases?

**Review by f-odhiambo — APPROVED** [review]
This works as expected

**Comment by ndegwamartin** [comment]
> @ndegwamartin are this available for prod releases?

It is not part of the current release but once it is merged it will be

**Review by pld — DISMISSED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt [hunk 1]
```diff
@@ -218,10 +218,17 @@ class RegisterFragment : Fragment(), OnSyncListener {
         // Show error message in snackBar message
         // syncJobStatus.exceptions may be null when worker fails; hence the null safety usage
         val hasAuthError =
-          syncJobStatus.exceptions.any {
-            it.exception is HttpException && (it.exception as HttpException).code() == 401
+          try {
+
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:222]: Delete extra space

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt [hunk 2]
```diff
@@ -218,10 +218,17 @@ class RegisterFragment : Fragment(), OnSyncListener {
         // Show error message in snackBar message
         // syncJobStatus.exceptions may be null when worker fails; hence the null safety usage
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:219]: We may need to re-word this comment

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt [hunk 3]
```diff
@@ -147,4 +150,28 @@ constructor(
...
+      withContext(dispatcherProvider.io()) {
+        val unsyncedResources =
+          fhirEngine
+            .getUnsyncedLocalChanges()
+            .groupingBy { it.localChange.resourceType.spaceByUppercase() }
+            .eachCount()
+            .map { it.key to it.value }
+
+        if (unsyncedResources.isNullOrEmpty()) {
+
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt:165]: Delete extra space

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**